### DISCUSSION TITLE
Get message by replyToMsgId

### DISCUSSION THREAD

**Discussion by marekkirillov** [gram-js/gramjs#549] [Q&A]
Hi! Does anyone know how to properly get the original message that is being replied to? I tried using the replyToMsgId and API.messages.GetMessages function, but it will always return a MessageEmpty object.

Some code:

Im using this handler to get notified when a new message arrives:
`client.addEventHandler(eventPrint, new NewMessage({ chats: [chatIds] }));`

This is the part from the eventPrint fn that should query the original message:
```
let replyToMsgId = event.message.replyTo?.replyToMsgId;
if (replyToMsgId) {
  const result = await client.invoke(
    new Api.({
      id: [replyToMsgId],
    })
  );
  console.log(result);
}
  ```
  
The result is always something like that:

```
{
      CONSTRUCTOR_ID: XXXXXXXXXXXXXX,
      SUBCLASS_OF_ID: XXXXXXXXXXXXXX,
      className: 'MessageEmpty',
      classType: 'constructor',
      flags: 0,
      id: <replyToMsgId>,
      peerId: null
    }
```

**Comment by rojvv** [comment]
You went too far. It is as easy as:

```ts
const repliedMessage = await event.message.getReplyMessage();
```
